experimental/GModule/Cohomology.jl

```diff
@@ -474,7 +474,7 @@ export action, cohomology_group, extension, pc_group_with_isomorphism
 export induce, is_consistent, istwo_cocycle, all_extensions
 export split_extension, extension_with_abelian_kernel
 
-Oscar.dim(C::GModule) = rank(C.M)
+Oscar.dim(C::GModule) = torsion_free_rank(C.M)
```

Comment:
```suggestion
Oscar.dim(C::GModule) = rank(C.M)
```
This delegates to the module and not the group. So the change here is wrong

Reply:
I'm not sure this is true. `C.M` is "the module", but there are some signatures with `C::GModule{..., FinGenAbGroup}` and the second type parameter is for `C.M` as far as I can see. Somebody who knows about the code should have a look.

Reply:
The CI in Hecke agrees that here we have a "module". See https://github.com/thofma/Hecke.jl/actions/runs/8094631614/job/22124040318#step:7:1375

Reply:
Oh, I think it can be both here. So we would need something like
```julia
Oscar.dim(C::GModule{..., FinGenAbGroup}) = torsion_free_rank(C.M)
Oscar.dim(C:GModule) = rank(C.M)
```

Reply:
Yes, exactly. This is an example from the tests:
```
julia> k, a = cyclotomic_field(5)
(Cyclotomic field of order 5, z_5)

julia> zk = maximal_order(k)
Maximal order of Cyclotomic field of order 5 
with basis AbsSimpleNumFieldElem[1, z_5, z_5^2, z_5^3]

julia> U, mU = unit_group_fac_elem(zk)
(Z/10 x Z, UnitGroup map of Factored elements over Cyclotomic field of order 5
)

julia> A, mA = automorphism_group(PermGroup, k)
(Permutation group of degree 4, Map: permutation group of degree 4 -> generic group of order 4 with multiplication table -> set of automorphisms of Cyclotomic field of order 5)

julia> C = gmodule(A, mU)
G-module for A acting on U

julia> typeof(C.M)
FinGenAbGroup

```